site/components/src/main/java/uk/nhs/digital/common/components/apispecification/ApiSpecificationRendererDirective.java

```diff
@@ -0,0 +1,54 @@
+package uk.nhs.digital.common.components.apispecification;
+
+import freemarker.core.Environment;
+import freemarker.template.*;
+import org.apache.commons.lang3.StringUtils;
+import org.hippoecm.hst.site.HstServices;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Optional;
+
+public class ApiSpecificationRendererDirective implements TemplateDirectiveModel {
+
+    private static final Logger log = LoggerFactory.getLogger(ApiSpecificationRendererDirective.class);
+
+    @Override public void execute(final Environment environment,
+                                  final Map parameters,
+                                  final TemplateModel[] loopVars,  // ignored
+                                  final TemplateDirectiveBody body // ignored
+    ) throws TemplateException, IOException {
+
+        final OpenApiSpecificationJsonToHtmlConverter converter
+            = HstServices.getComponentManager().getComponent("apiSpecificationRenderer");
+
+        final String specificationJson = stringParam(parameters, "specificationJson").filter(StringUtils::isNotBlank).orElse(null);
+
+        final String path = stringParam(parameters, "path").orElse("not available");
+
+        if (specificationJson == null) {
+            log.debug("No API specification JSON available for {}", path);
+            return;
+        }
+
+        try {
+            log.debug("Rendering API specification {}: start", path);
+            final String specificationHtml = converter.htmlFrom(specificationJson);
+            log.debug("Rendering API specification {}: done", path);
+
+            environment.getOut().append(specificationHtml);
+
+        } catch (final RuntimeException | IOException e) {
```

Comment:
i've never seen the bitwise OR operator used in this way, could you explain how it works? moreso i can use it in the future, rather than understand it here

Reply:
Keen eye!

It's a shorthand for [handling exceptions of various types in a single `catch` clause](https://docs.oracle.com/javase/7/docs/technotes/guides/language/catch-multiple.html), rather than having one catch clause per exception type.

Reply:
now here i was hoping i'd somehow missed [union types](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types) being introduced to java :joy: